src/Traces.Web/Services/ApaleoSetupService.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Traces.ApaleoClients.Integration.Models;
+using Traces.Common;
+using Traces.Common.Constants;
+using Traces.Common.Enums;
+using Traces.Common.Exceptions;
+using Traces.Common.Utils;
+using Traces.Core.ClientFactories;
+
+namespace Traces.Web.Services
+{
+    internal class ApaleoSetupService : IApaleoSetupService
+    {
+        private readonly IApaleoIntegrationClientFactory _apaleoIntegrationClientFactory;
+        private readonly IOptions<IntegrationConfig> _integrationConfig;
+
+        private readonly Dictionary<string, ApaleoIntegrationTargetsEnum> apaleoIntegrationTargetDictionary =
+            new Dictionary<string, ApaleoIntegrationTargetsEnum>
+            {
+                { ApaleoOneConstants.AccountMenuAppsIntegrationCode, ApaleoIntegrationTargetsEnum.AccountMenuApps },
+                { ApaleoOneConstants.PropertyMenuAppsIntegrationCode, ApaleoIntegrationTargetsEnum.PropertyMenuApps },
+                { ApaleoOneConstants.ReservationDetailsTabIntegrationCode, ApaleoIntegrationTargetsEnum.ReservationDetailsTab }
+            };
+
+        public ApaleoSetupService(IApaleoIntegrationClientFactory apaleoIntegrationClientFactory, IOptions<IntegrationConfig> integrationConfig)
+        {
+            _integrationConfig = Check.NotNull(integrationConfig, nameof(integrationConfig));
+            _apaleoIntegrationClientFactory = Check.NotNull(apaleoIntegrationClientFactory, nameof(apaleoIntegrationClientFactory));
+        }
+
+        public async Task<bool> SetupApaleoUiIntegrationsAsync()
```

Comment:
```suggestion
        public async Task SetupApaleoUiIntegrationsAsync()
```

you just return true or throw exceptions, then you might as well omit the return value